src/test/java/com/exactpro/epfast/DecoderUtils.java

```diff
@@ -9,9 +9,16 @@
 
     public static void decode(IDecodeContext decoder, Iterable<ByteBuf> buffers) {
         Iterator<ByteBuf> it = buffers.iterator();
-        decoder.decode(it.next());
-        while (!decoder.isReady() && it.hasNext()) {
-            decoder.continueDecode(it.next());
+        decoder.decode(nextNonEmptyBuffer(it));
+    }
+
+    private static ByteBuf nextNonEmptyBuffer(Iterator<ByteBuf> buffers) {
+        while (buffers.hasNext()) {
+            ByteBuf buffer = buffers.next();
+            if (buffer != null && buffer.isReadable()) {
+                return buffer;
+            }
         }
+        return null;
```

Comment:
It's good place to throw Execption saying that there are no non-empty buffers left